aten/src/ATen/native/group_norm.cpp

```diff
@@ -120,11 +120,15 @@ std::tuple<Tensor, Tensor, Tensor> native_group_norm_backward(
       at::borrow_from_optional_tensor(gamma_opt);
   const Tensor& gamma = *gamma_maybe_owned;
   TORCH_CHECK(
-      X.suggest_memory_format() == dY.suggest_memory_format(),
-      "Expected memory formats of X and dY are same.");
+      X.device().is_cpu() ?
```

Comment:
Yes, CUDA should also have same memory_format for X and dY while XPU doesn't need to have same memory_format.  https://github.com/intel/intel-extension-for-pytorch/blob/xpu-master/csrc/gpu/aten/operators/GroupNorm.cpp#L525.

Reply:
Actually, I think we can make a direct check because XPU's `native_group_norm_backward` will replace `native_group_norm_backward` of torch and there should be no assertion failure https://github.com/intel/intel-extension-for-pytorch/blob/xpu-master/scripts/tools/torchgen/yaml/xpu_functions.yaml#L434 . Have you encountered any problems caused by this check ?

Reply:
Apologies for the delay, here is a repro script for a case that errors on CPU

```
import torch
import torch.nn as nn
N = 2048
C = 384
H = 1
W = 1
# separate 384 channels into 3 groups
groups = 3
input = torch.randn(H, N, C, W, requires_grad=True, device='cpu').permute(1, 2, 0, 3)
print(f"input shape: {input.shape}, stride: {input.stride()}")
print(input.is_contiguous(memory_format=torch.channels_last))
print(input.stride())
m = nn.GroupNorm(groups, C, affine=False, device='cpu')
output = m(input)
gradient = torch.randn(N, C, H, W, device='cpu').to(memory_format=torch.channels_last)
print(f"gradient shape: {gradient.shape}, stride: {gradient.stride()}")
output.backward(gradient) # RuntimeError: Expected memory formats of X and dY are same.
```

Reply:
Remove memory format check for the edge cases https://github.com/pytorch/pytorch/issues/115940.

Reply:
Agreed that we can remove this check.

> Although `suggest_memory_format()` will return different values for X and dY

Could you elaborate more on why this happens?

Reply:
> why suggest_memory_format() will return different values for X and dY

It is because `is_channels_last_strides_2d_s4` will check the stride for the dim with size = 1. While `_compute_channels_last_contiguous_2d` and `_compute_contiguous` will skip the check for dims with size=1.
Take https://github.com/pytorch/pytorch/issues/115940 as instance, X is contiguous and also channels_last contiguous but it is not channels_last. dY is contiguous and also channels_last contiguous, and it is also channels last.

It is not a bug, and we should be aware that channels_last and channels_last_contiguous may be inconsistent.